### PR TITLE
Fix lint issues

### PR DESCRIPTION
**Pull request by khos2ow**
Fixes lint issues:

- (golint) comment on exported const
- (errcheck) error return value not checked
- (vet) result of fmt.Errorf call not used
- (staticcheck) error strings should not be capitalized
- (staticcheck) should replace loop
- Remove obsolete linters: gosimple, unused

### CODE REVIEW

File: cloudca/resource_cloudca_load_balancer_rule.go [hunk 1]
```diff
@@ -145,17 +145,17 @@ func readLbr(d *schema.ResourceData, meta interface{}) error {
...
-	d.Set("network_id", lbr.NetworkId)
-	d.Set("instance_ids", lbr.InstanceIds)
-	d.Set("algorithm", lbr.Algorithm)
-	d.Set("protocol", lbr.Protocol)
-	d.Set("public_port", lbr.PublicPort)
-	d.Set("private_port", lbr.PrivatePort)
-	d.Set("public_ip", lbr.PublicIp)
-	d.Set("stickiness_method", lbr.StickinessMethod)
-	d.Set("stickiness_params", lbr.StickinessPolicyParameters)
+	_ = d.Set("name", lbr.Name)
```
Comment (franzpgarcia) [inline cloudca/resource_cloudca_load_balancer_rule.go]: This is kind of ugly but I guess if the linter is asking for it then I have no problem with that.

**Comment by khos2ow** [comment]
@pdube @franzpgarcia I agree with both of you and basically added `_ = fmt.Errorf` to make linter happy, and asides from Pat's suggestion we can also ignore them on lint level. Both approaches seem good to me. Do you have any preferences between the two?

**Comment by pdube** [comment]
I would prefer to not have any lint ignores

**Comment by khos2ow** [comment]
According to another provider as an example we might want to do something like this:

```

**Comment by khos2ow** [comment]
@pdube @franzpgarcia this should be good now.

File: cloudca/resource_cloudca.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ func readIntFromString(valStr string) int {
 func handleNotFoundError(err error, d *schema.ResourceData) error {
 	if ccaError, ok := err.(api.CcaErrorResponse); ok {
 		if ccaError.StatusCode == 404 {
-			fmt.Errorf("Entity (id=%s) not found", d.Id())
+			_ = fmt.Errorf("Entity (id=%s) not found", d.Id())
```
Comment (pdube) [inline cloudca/resource_cloudca.go]: Sorry if I wasn't clear enough before. Lines like these are useless. It creates a string then it is ignored. It doesn't log it or anything. We should either remove the lines or log them at debug level.

Comment (khos2ow) [reply cloudca/resource_cloudca.go]: In 20e3b98 I reverted and applied error handling for all the rest, but these ones were never actually logged and printed before. That's why I didn't touch them.
I'm going to bubble these up too.

Comment (pdube) [reply cloudca/resource_cloudca.go]: Ok cool. Be careful though, in some cases a not found may be acceptable (e.g. delete instance, but instance is not found)

Comment (khos2ow) [reply cloudca/resource_cloudca.go]: 83c5b00 unifies this behavior across all files.

**Pull request merged**